src/components/targets/TargetForm.js

```diff
@@ -27,16 +28,21 @@ class TargetForm extends React.Component{
     updateTargetInfo({topic: index});
   }
 
+  onTargetDelete(e){
+    e.preventDefault();
+
+    const { deleteTargetAction, currentTarget } = this.props;
```

Comment:
You are use only the `id` of the `currentTarget`, so you can use a shortcut here:
`const { deleteTargetAction, currentTarget: { id } } = this.props;`